### PR TITLE
Avoid joint jump when SuddenHalt() is called in velocity mode

### PR DESCRIPTION
**Pull request by nbbrooks**
### Description

[This is a backport of a MoveIt2 bug fix](https://github.com/ros-planning/moveit2/pull/410/files)

We found out the hard way, when the suddenHalt() function was called for a robot in velocity control mode, it would send a command vector of all-zero positions. That could cause a really big joint jump depending what type of ros2_controller is active. This fixes that.

Also (really minor fix) it sets trajectory accelerations to 0 to match the comment.

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by v4hn** [comment]
Hi Nathan,

thanks for backporting. This is clearly urgent and for me would even justify a quick followup release for noetic because it's a safety issue. @tylerjw 

The PR fails CI because of MoveIt/MoveIt2 divergence:
Apparently [this commit](https://github.com/ros-planning/moveit2/commit/db1f3227ea59af9aac3cd290a0ce4555eddc3718#diff-7355431db583f9f75319bb82dee0c9379da3a7a720e37b49f07fc9e80d6175a9R243) switched to pointers (the I have no idea how this can be a "Port from MoveIt" commit when MoveIt does not even have these changes...)
I guess you could just change to `.` member access to solve it.

**Comment by AndyZe** [comment]
It's still pretty unlikely this would cause issues. Somebody would need to be using a controller that accepts optional positions & vels, but only have Servo configured to publish vels. Definitely unusual!

But yeah, we should get it merged.

**Comment by tylerjw** [comment]
> I have no idea how this can be a "Port from MoveIt" commit when MoveIt does not even have these changes...

It is a ROS1 -> ROS2 port.  The code that it is modifying was a copy of the ros1 code and was previous to this PR explicitly ignored.  There are some new features also (listed in the text of the commit message) that we also added like some more tests and examples.

As to why this variable changed to a pointer type, (if I remember correctly) that has to do with changes needed to support the changes in the parameter system from ROS2.

**Comment by nbbrooks** [comment]
Oops, fixed the pointer access issue.

**Comment by tylerjw** [comment]
Just like in another PR @AndyZe was asking me about.  You need to fetch and rebase this on master to get it to pass CI.  The name of the `Format` check was changed to `Format / pre-commit` and so the required `Format` check will never run.

tl;dr rebase on master.

**Comment by v4hn** [comment]
> tl;dr rebase on master.

The bigger problem was that `Format` was still a required CI job (which never ran).
I adjusted the github repo settings just now and made `pre-commit` the required job.

**Review by v4hn — APPROVED** [review]
